src/kube_vxlan_controller_cli.erl

```diff
@@ -71,17 +69,41 @@
 
 -define(Version, "Version ~s (git-~s)~n").
 
-args(AllArgs) -> case AllArgs of
-    ["run"|Args] -> {run, read_args(Args)};
-    ["list", Subject|Args] -> {list, list_to_atom(Subject), read_args(Args)};
-    ["inspect", Subject|Args] ->
-        {inspect, list_to_atom(Subject), read_args(Args)};
-    ["config"|Args] -> {config, read_args(Args)};
-    ["version"] -> version;
-    ["help", Command|_Args] -> {help, list_to_atom(Command)};
-    [Command, "help"|_Args] -> {help, list_to_atom(Command)};
-    _Other -> usage
-end.
+cli(CliArgs) ->
+    case args(CliArgs) of
+	{list, Subject, Args} -> do({list, Subject}, Args);
+	{inspect, Subject, Args} -> do({inspect, Subject}, Args);
+	{config, Args} -> do(config, Args);
+	{help, Command} -> io:format(help(Command));
+	version -> io:format(version(?Config:version()));
+	usage -> io:format(usage())
+    end.
+
+do(config, {NamedArgs, _OrderedArgs}) ->
+    io:format("~p~n", [?Config:init()]),
+    io:format("~p~n", [?Config:read(NamedArgs)]);
+
+do(Action, {NamedArgs, OrderedArgs}) ->
+    case ?Config:load(NamedArgs) of
+	{ok, Config} -> do(Action, OrderedArgs, Config);
+	{error, Reason} -> io:format("~p~n", [Reason])
+    end.
+
+do({list, Subject}, Args, Config) -> ?List:Subject(Args, Config);
+do({inspect, Subject}, Args, Config) -> ?Inspect:Subject(Args, Config).
+
+args(AllArgs) ->
+    case AllArgs of
+	["run"|Args] -> {run, read_args(Args)};
```

Comment:
The caller of args/1 does not expect return value {run, ...}

Reply:
yeah, that was moved from the old main function. A few things that are obsolete now creeped in.
I probably broke the command line support anyway. If this is really needed/useful, a new task/PR should be created for that.